hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/Pipelines.java

```diff
@@ -485,9 +485,10 @@ public static DataStreamSink<Object> clean(Configuration conf, DataStream<Object
     return cleanCommitDataStream;
   }
 
-  public static DataStreamSink<Object> dummySink(DataStream<Object> dataStream) {
+  public static DataStreamSink<Object> dummySink(Configuration conf, DataStream<Object> dataStream) {
     return dataStream.addSink(Pipelines.DummySink.INSTANCE)
-        .setParallelism(1)
+        // sets the same parallelism to upstream operators to enable partial failover.
+        .setParallelism(conf.getInteger(FlinkOptions.WRITE_TASKS))
```

Comment:
Don't need to set the parallelism explicitly, the parallelism will be same as  that of upstream write operator if not set, and will automatically chained together.